commands/logging-service/pipeline/create.go

```diff
@@ -0,0 +1,131 @@
+)
+
+func PipelineCreateCmd() *core.Command {
+	cmd := core.NewCommandWithJsonProperties(
+		context.Background(), nil, jsonPropertiesExample, &pipelineToCreate, core.CommandBuilder{
+			Namespace: "logging-service",
+			Resource:  "pipeline",
+			Verb:      "create",
+			ShortDesc: "Create a logging pipeline",
+			Example: `ionosctl logging-service pipeline create --json-properties PATH_TO_FILE
+ionosctl logging-service pipeline create --json-properties-example
+ionosctl logging-service pipeline create --name NAME --log-tag LOG_TAG --log-source LOG_SOURCE --log-protocol
+LOG_PROTOCOL --log-retention-time LOG_RETENTION_TIMES`,
+			PreCmdRun: preRunCreateCmd,
+			CmdRun:    runCreateCmd,
+		},
+	)
+	cmd.Command.Flags().StringSlice(constants.ArgCols, defaultCols, tabheaders.ColsMessage(defaultCols))
+	cmd.AddStringFlag(
+		constants.FlagName, constants.FlagNameShort, "", "Sets the name of the pipeline",
+	)
+	cmd.AddStringFlag(
+		constants.FlagLoggingPipelineLogTag, "", "", "Sets the tag for the pipeline log",
+	)
+	cmd.AddSetFlag(
+		constants.FlagLoggingPipelineLogSource, "", "", []string{"docker", "systemd", "generic", "kubernetes"},
+		"Sets the source for the pipeline log",
+	)
+	cmd.AddSetFlag(
+		constants.FlagLoggingPipelineLogProtocol, "", "", []string{"http", "tcp"},
+		"Sets the protocol for the pipeline log",
+	)
+	cmd.AddStringSliceFlag(constants.FlagLoggingPipelineLogLabels, "", nil, "Sets the labels for the pipeline log")
+	cmd.AddStringFlag(
+		constants.FlagLoggingPipelineLogType, "", "loki",
+		"Sets the destination type for the pipeline log",
+	)
+	cmd.AddSetFlag(
+		constants.FlagLoggingPipelineLogRetentionTime, "", "30", []string{"7", "14", "30"},
+		"Sets the retention time in days for the pipeline log",
+	)
+
+	return cmd
+}
+
+func runCreateCmd(c *core.CommandConfig) error {
```

Comment:
If I understand correctly, unless the the name of the pipeline is specified, the default behavior is to create an empty pipeline ?

Reply:
No, there are 3 ways to use the create command:

- with the `--json-properties-example` flag, which will print an example for the json properties file that should be used for creating a pipeline (it doesn't actually run the create command)
- with the `--json-properties` flag, which takes the path to a file where the properties of the pipeline have already been defined (the parsing of the file is done in a separate function that was not added by this PR)
- with the required flags (`--name`, `--log-tag`, `--log-source`, `--log-protocol`) and there are some flags which can be populated, but are not required. This will create a pipeline with **only** 1 log

I did not, however, take into consideration the possibility of a user setting both `--json-properties` and `--name`, for instance, so I will look into making that check more clear